### PR TITLE
Make mouse wheel interaction runtime configurable

### PR DESCRIPTION
**Pull request by Antreasgr**
Added maxdelta, timeout runtime configurable parameters in
ol.interaction.MouseWheelZoom class according to
http://stackoverflow.com/questions/28888047/mouse-wheel-scroll-limitation-in-openlayers-3

### CODE REVIEW

File: externs/olx.js [hunk 1]
```diff
@@ -2581,6 +2581,21 @@ olx.interaction.MouseWheelZoomOptions.prototype.duration;
...
+ */
+olx.interaction.MouseWheelZoomOptions.prototype.maxdelta;
+
+/**
+ * Timeout duration. Default is `80`.
+ * @type {number}
+ * @api
+ */
+olx.interaction.MouseWheelZoomOptions.prototype.timeout;
+
```
Comment (bartvde) [inline externs/olx.js]: you need to add these new options here as well:

```
/**
 * @typedef {{duration: (number|undefined)}}
 * @api
 */
olx.interaction.MouseWheelZoomOptions;
```

**Comment by ahocevar** [comment]
@bartvde Are you going to continue reviewing this, or should someone else take over?

**Comment by bartvde** [comment]
This probably needs updating not use goo.match.clamp right? https://github.com/openlayers/ol3/pull/4175

**Comment by waissbluth** [comment]
Is anybody on this? This feature would be really useful.

**Comment by rixlabs** [comment]
+1

**Comment by tschaub** [comment]
I like the idea of making this configurable.  But I think it is more important that we have better mouse wheel interaction by default.  Based on the [StackOverflow question](http://stackoverflow.com/questions/28888047/mouse-wheel-scroll-limitation-in-openlayers-3), I assume we'll have differing opinions of "better" though.  In my opinion, with a trackpad on a MacBook, the map zooms in/out way more than I want it to every time.  Based on experience watching people try to use OpenLayers for the first time, I think it is common to be surprised by the excessive zooming.

**Comment by rixlabs** [comment]
I wish I could configure the interaction to avoid some eccessive call to my map services.
I noticed the problem cause I have poor arcgis test environment and with default config It happens that OL make several call for a mousewheel zoom that cause(my server is really under size) some timeout problems form time to time. To solve it I had to change the 80ms time to 250 going into the code (ol.js)

**Comment by tschaub** [comment]
If the remaining work and conflict resolution can be completed before the upcoming 4.6 release, please reopen this pull request.  After 4.6, we will be transforming the sources to ES modules, and if we want to pick up this effort, it may be easier to create the changes from scratch.

**Pull request closed without merging**